services/hh/processor.go

```diff
@@ -284,3 +306,41 @@ func (p *Processor) PurgeOlderThan(when time.Duration) error {
 	}
 	return nil
 }
+
+func (p *Processor) DeleteOlderThan(when time.Duration) error {
+	p.mu.Lock()
+	defer p.mu.Unlock()
+
+	deletedQueues := make([]uint64, 0)
+	for nodeID, queue := range p.queues {
+		// Only delete queues for inactive nodes.
+		ni, err := p.metastore.Node(nodeID)
+		if err != nil {
+			return err
+		}
+		if ni != nil {
+			continue
+		}
+
+		last, err := queue.LastModified()
+		if err != nil {
+			return err
+		}
+		if last.Before(time.Now().Add(-when)) {
+			// Close and remove the queue.
+			if err := queue.Close(); err != nil {
+				return err
+			}
+			if err := os.RemoveAll(queue.dir); err != nil {
```

Comment:
I'd say return an error if you call `Remove` and the queue is not already closed.